float32 velocity
---
bool result
